### DISCUSSION TITLE
Support for "Teams" for database

### DISCUSSION THREAD

**Discussion by saturnonearth** [supabase/supabase#7562] [Feature Requests]
I am creating an app that will require support for "teams" where certain users will only have read access to certain database data. It would be cool if we could set this up directly within the database and auth for Supabase.

Appwrite has this feature, see here for more info:
[https://appwrite.io/docs/client/teams](https://appwrite.io/docs/client/teams)

I'm sure you can do this without this manually without this type of feature, but from my understanding, it seems like this would make it easier to do.

Hopefully I am not confused as to what this feature actually entails.

**Comment by GaryAustin1** [comment]
This is all possible, but you there is not built in support.  If you search teams here you might find some info, or Google Postgres and teams there might be examples of setting up the table structure.  There are also examples of departments with teams and individual. 

With RLS you can certainly enforce access only to certain tables with update/insert privileges based on role and membership in a team.

This link has suggestions and code to use auth to support the roles in the jwt to avoid table lookups in the RLS.  https://github.com/supabase-community/supabase-custom-claims
